lib/manageiq/api/common/user.rb

```diff
@@ -0,0 +1,87 @@
+module ManageIQ
+  module API
+    module Common
+      class User
+        IDENTITY_KEY = 'x-rh-identity'.freeze
+
+        def self.current
+          Thread.current[:attr_current_user]
+        end
+
+        def self.current=(user_header)
+          Thread.current[:attr_current_user] = new(user_header)
+        end
+
+        def self.with_logged_in_user(user_header)
+          self.current = user_header
+          yield self.current
+        ensure
+          Thread.current[:attr_current_user] = nil
```

Comment:
@syncrou 
Should this be self.current = nil instead of directly setting the TLS?

Reply:
I can call `self.current = nil` but I have to then check that method to see if nil is passed in.  Otherwise new is always called. Do we want `self.current` to have the ability to be set as `nil`?

Reply:
It feels to me like the `current=` setter maybe should not take a raw header, because it's inconsistent with the getter which returns a wrapped object. If we make the getter return a wrapped object, and the setter take a wrapped object, then we either need another setter (like `raw_current=` and/or some kind of factory method `User.from_header`).  Once all that's in place, `current=nil` should be straightforward.